codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -2061,47 +2084,9 @@ runStatic pic = do
 
 -- Wraps the event and state from run so they can be paused by pressing the Inspect
 -- button.
-runInspect ::
-       s -> (Double -> s -> s) -> (Event -> s -> s) -> (s -> Picture) -> IO ()
-runInspect initial stepHandler eventHandler drawHandler = do
-    Just window <- currentWindow
-    Just doc <- currentDocument
-    Just canvas <- getElementById doc ("screen" :: JSString)
-    let initialWrapper = (debugStateInit, initial)
-        stepHandlerWrapper dt (debugState, state) =
-            case debugStateActive debugState of
-                True -> (debugState, state)
-                False -> (debugState, stepHandler dt state)
-        eventHandlerWrapper evt (debugState, state) =
-            case evt of
-                Left debugEvent ->
-                    (updateDebugState debugEvent debugState, state)
-                Right normalEvent ->
-                    (debugState, eventHandler normalEvent state)
-        drawHandlerWrapper (debugState, state) =
-            drawDebugState debugState (pictureToDrawing $ drawHandler state)
-        drawPicHandler (debugState, state) = drawHandler state
-    (sendEvent, getState) <-
-        run
-            initialWrapper
-            stepHandlerWrapper
-            eventHandlerWrapper
-            drawHandlerWrapper
-            (Right . TimePassing)
-    let pauseEvent True = sendEvent $ Left DebugStart
-        pauseEvent False = sendEvent $ Left DebugStop
-        highlightSelectEvent True n = sendEvent $ Left (HighlightEvent n)
-        highlightSelectEvent False n = sendEvent $ Left (SelectEvent n)
-    onEvents canvas (sendEvent . Right)
-    inspect (drawPicHandler <$> getState) pauseEvent highlightSelectEvent
-
-runPauseable ::
-       Wrapped s
-    -> (Double -> s -> s)
-    -> (Wrapped s -> [Control s])
-    -> (s -> Picture)
-    -> IO ()
-runPauseable initial stepHandler controls drawHandler = do
+runInspect :: 
```

Comment:
This type signature would make a lot more sense to me if the second parameter was an `s` rather than a `Wrapped s`.  You can do that by moving the initialization of Wrapped into this function.  I'd like to see that happen if you want to do it.  Otherwise, I'll do it myself after the merge.

(This isn't a new problem.  The type of the old `runPausable` was already pretty weird.)

Reply:
I'd like to understand how this works. I can try to work on this tomorrow and see if I can get it to work (ie before we meet).